shell/server/installer.js

```diff
@@ -389,7 +387,12 @@ AppInstaller = class AppInstaller {
             (manifest.appMarketingVersion && manifest.appMarketingVersion.defaultText));
         }
 
-        if (_this.isPreinstalled) {
+        if (globalDb.getPackageIdForPreinstalledApp(_this.appId) &&
+            globalDb.collections.appIndex.findOne({ "packageId": _this.packageId })) {
```

Comment:
Nitpick: This query would be more efficient if you searched by _id as well. Otherwise it's going to table-scan.